eidas-saml-parser/src/main/java/uk/gov/ida/notification/eidassaml/EidasSamlResource.java

```diff
@@ -72,7 +74,12 @@ public EidasSamlParserResponse post(@Valid EidasSamlParserRequest request) throw
         return new EidasSamlParserResponse(
                 authnRequest.getID(),
                 authnRequest.getIssuer().getValue(),
-                assertionConsumerServiceURL);
+                assertionConsumerServiceURL,
+                assignTransientPid(authnRequest));
+    }
+
+    private boolean assignTransientPid(AuthnRequest authnRequest) {
+        return authnRequest.getNameIDPolicy() != null && NameID.TRANSIENT.equals(authnRequest.getNameIDPolicy().getFormat()) ? true : false;
```

Comment:
Don't think we need the ternary - the expression up to the ternary returns the correct boolean - and your tests agree :)